components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/SessionDataStore.java

```diff
@@ -65,35 +67,41 @@
     /**
      * Returning instance of one of the implementation of SessionDataStore.
      *
-     * @return {@link RdbmsSingleEntrySessionDataStore} or {@link RedisSessionDataStore}
+     * @return {{@link SessionDataStore}
      */
     public static SessionDataStore getInstance() {
 
-        String enablePersistVal = IdentityUtil.getProperty("JDBCPersistenceManager.SessionDataPersist.Enable");
-        enablePersist = true;
-        if (enablePersistVal != null) {
-            enablePersist = Boolean.parseBoolean(enablePersistVal);
-        }
-
-        String redisEnabledVal = IdentityUtil.getProperty("JDBCPersistenceManager.SessionDataPersist.RedisEnable");
-        if (StringUtils.isNotBlank(redisEnabledVal)) {
-            redisEnabled = Boolean.parseBoolean(redisEnabledVal);
-        }
         if (instance == null) {
             synchronized (SessionDataStore.class) {
                 if (instance == null) {
-                    if (redisEnabled == true) {
-                        instance = new RedisSessionDataStore();
-                    } else {
-                        instance = new RdbmsSingleEntrySessionDataStore();
+                    instance = DEFAULT_SESSION_DATA_STORE;
+                    if (FrameworkServiceDataHolder.getInstance().getSessionDataStores().containsKey(getImplType())) {
+                        instance = FrameworkServiceDataHolder.getInstance().getSessionDataStores().get(getImplType());
```

Comment:
Here let's check if `getImpleType()` is not empty or null. If its emtpy or null, lets use the default session store name. Ideally that default one should be deployed in all scenarios. 

Because of that we dont need to instantiate a default object here